### PR TITLE
Add siguza-utils

### PR DESCRIPTION
**Pull request by TheRealKeto**
This PR adds ``siguza-utils``, which includes ``bindump``, ``clz``, ``dsc_syms``, ``rand``, ``strerror``, ``vmacho``, and ``xref`` from [Siguza/misc](https://github.com/Siguza/misc), in one package. The only exception is ``mesu``, which seems to no longer work.

This is my first, somewhat usable package (that isn't just a control file), so I do suggest giving me criticism on how I've compiled this. I've attempted to use functions which make things like downloading and extracting a package easier, but I'm sure this can be improved. Software provided as-is, blah blah not much from here will do anything bad.

### CODE REVIEW

File: siguza-utils.mk [hunk 1]
```diff
@@ -0,0 +1,85 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUPROJECTS      += siguza-utils
```
Comment (Diatrus) [inline siguza-utils.mk:5]: No need to have a LIBS var for just one tool. Just put the libs/frameworks after the ldflags.

File: build_info/siguza-utils.control [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Package: siguza-utils
+Version: @DEB_MISC_V@
+Architecture: @DEB_ARCH@
+Maintainer: @DEB_MAINTAINER@
+Author: Siguza
+Section: Utilities
+Priority: optional
+Tag: purpose::console, role::hacker
```
Comment (Diatrus) [inline build_info/siguza-utils.control:8]: I try to avoid using Tag: as of late.

**Review by Diatrus — CHANGES_REQUESTED** [review]
Overall looks great, just a couple surface-level quirks I see from a quick scan. Nice description btw.

File: build_info/siguza-utils.control [hunk 3]
```diff
@@ -5,7 +5,6 @@ Maintainer: @DEB_MAINTAINER@
 Author: Siguza
 Section: Utilities
 Priority: optional
-Tag: purpose::console, role::hacker
```
Comment (Cryptiiiic) [inline build_info/siguza-utils.control]: no one likes u get lost

File: siguza-utils.mk [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+# Don't change the version, just the date and git hash
+MISC_COMMIT     := 43852780a20ebf2c990681aa1ab42577ae630555
+MISC_VERSION    := 1.0+git20210410.$(shell echo $(MISC_COMMIT) | cut -c -7)
+DEB_MISC_V      ?= $(MISC_VERSION)
+
+siguza-utils-setup: setup
+	$(call GITHUB_ARCHIVE,Siguza,misc,$(MISC_COMMIT),$(MISC_COMMIT))
+	$(call EXTRACT_TAR,misc-$(MISC_COMMIT).tar.gz,misc-$(MISC_COMMIT),misc)
```
Comment (CRKatri) [inline siguza-utils.mk:13]: Change arg 3 from `misc` to `siguza-utils` and delete line 23

File: siguza-utils.mk [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+endif
+
+SUPROJECTS      += siguza-utils
+# Don't change the version, just the date and git hash
+MISC_COMMIT     := 43852780a20ebf2c990681aa1ab42577ae630555
+MISC_VERSION    := 1.0+git20210410.$(shell echo $(MISC_COMMIT) | cut -c -7)
+DEB_MISC_V      ?= $(MISC_VERSION)
+
+siguza-utils-setup: setup
+	$(call GITHUB_ARCHIVE,Siguza,misc,$(MISC_COMMIT),$(MISC_COMMIT))
```
Comment (CRKatri) [inline siguza-utils.mk:12]: Use arg 4 when changing the name of the tarball (look at ghostbin for reference)

**Review by CRKatri — CHANGES_REQUESTED** [review]

File: build_info/siguza-utils.control (hunk 3 above)
Comment (TheRealKeto) [reply build_info/siguza-utils.control]: this is a banger

**Comment by TheRealKeto** [comment]
The latest 2 commits should accommodate with all requested changes.

File: siguza-utils.mk [hunk 6]
```diff
@@ -0,0 +1,82 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUPROJECTS      += siguza-utils
+# Don't change the version, just the date and git hash
+MISC_COMMIT     := 43852780a20ebf2c990681aa1ab42577ae630555
```
Comment (Diatrus) [inline siguza-utils.mk:7]: name these SIGUZA-UTILS_{COMMIT,VERSION,etc} for consistency please

**Review by Diatrus — CHANGES_REQUESTED** [review]

**Pull request merged**